src/cargo/ops/cargo_rustc/fingerprint.rs

```diff
@@ -342,6 +351,7 @@ fn calculate<'a, 'cfg>(cx: &mut Context<'a, 'cfg>, unit: &Unit<'a>)
         deps: deps,
         local: local,
         memoized_hash: Mutex::new(None),
+        rustflags: env::var("RUSTFLAGS").ok(),
```

Comment:
Could this use the methods on the context instead? That way if you use `RUSTFLAGS` with `--target` you won't recompile any build scripts or plugins (as they're not affected by this env var anyway)